src/Microsoft.AspNet.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -89,7 +88,8 @@ public OAuthHandler(HttpClient backchannel)
                 var value = query["error"];
                 if (!StringValues.IsNullOrEmpty(value))
                 {
-                    Logger.LogVerbose("Remote server returned an error: " + Request.QueryString);
+                    Logger.LogWarning("Remote server returned an error: " + value);
```

Comment:
When calling `Redirect`, both the status code and the `Location` header are correctly updated, but since the execution is not stopped (despite the `context.HandleResponse()`), the rest of the pipeline is invoked and the default 404 handler modifies the 302 response: you end up getting a 404 response with a `Location` header that has no effect :smile: